src/dom/node.ts

```diff
@@ -340,10 +340,29 @@ export class Node extends EventTarget {
     if (this.parentNode) {
       const parentNode = this.parentNode;
       const mutator = parentNode._getChildNodesMutator();
-      const index = mutator.indexOf(this);
+      let viableNextSibling: Node | null = null;
+      {
+        const thisIndex = mutator.indexOf(this);
+        for (let i = thisIndex + 1; i < mutator.arrayInstance.length; i++) {
```

Comment:
The fact that `mutator` is backed by `parent.childNodes` is crucial to the behavior of this function, and the fact that the same array is accessed from both variables might be somewhat confusing, which is why I made `arrayInstance` a public field.

I've changed it to use `parent.childNodes` anyway.

Reply:
The `arrayInstance` field would need to be of type `NodeList` btw, because `.childNodes` isn't a normal array: it lacks standard array methods like `.map`, `.filter`, etc. The idea of the `mutator` is that it merely makes some array methods available on a `NodeList` through its mutator interface. The fact that it's also tied to a specific array (`.childNodes` being the array in question) is perhaps an interface design error on my part. 

In my opinion, exposing `arrayInstance` that way gives the false impression that there is a separate array instance, which is why I'm opposed to that specific change. Though if you have any ideas for a better design I'm open to hearing them.